packages/fxa-settings/src/models/pages/signin/query-params.ts

```diff
@@ -20,4 +20,24 @@ export class SigninQueryParams extends ModelDataProvider {
   @IsBoolean()
   @bind()
   hasPassword: boolean | undefined = undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  service: string | undefined = undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  redirectToUrl: string | undefined = undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  redirectPathname: string | undefined = undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  verificationReason: string | undefined = undefined;
```

Comment:
Why do we need this in a query param?

Reply:
Sorry, this was just me tinkering with passing that. I'm using location state now, and have cleaned this up.